tests/model_monitoring/test_store.py

```diff
@@ -0,0 +1,183 @@
+# Copyright 2024 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import string
+import time
+import typing
+from random import choice, randint
+from typing import Optional
+
+import pytest
+
+import mlrun.common.schemas
+import mlrun.model_monitoring
+import mlrun.model_monitoring.db.stores.sqldb.sql_store
+from mlrun.common.schemas.model_monitoring import WriterEvent
+from mlrun.model_monitoring.db.stores import (  # noqa: F401
+    StoreBase,
+)
+from mlrun.model_monitoring.writer import _AppResultEvent
+
+SQLstoreObject = typing.TypeVar("SQLstoreObject", bound="StoreBase")
+
+
+TEST_PROJECT = "test_model_endpoints"
+STORE_CONNECTION = "sqlite:///test.db"
+
+
+def _mock_random_endpoint(
+    state: Optional[str] = None,
+) -> mlrun.common.schemas.ModelEndpoint:
+    def random_labels():
+        return {f"{choice(string.ascii_letters)}": randint(0, 100) for _ in range(1, 5)}
+
+    return mlrun.common.schemas.ModelEndpoint(
+        metadata=mlrun.common.schemas.ModelEndpointMetadata(
+            project=TEST_PROJECT, labels=random_labels(), uid="some-ep-id"
+        ),
+        spec=mlrun.common.schemas.ModelEndpointSpec(
+            function_uri=f"test/function_{randint(0, 100)}:v{randint(0, 100)}",
+            model=f"model_{randint(0, 100)}:v{randint(0, 100)}",
+            model_class="classifier",
+        ),
+        status=mlrun.common.schemas.ModelEndpointStatus(state=state),
+    )
+
+
+@pytest.fixture(params=[0])
+def event(request: pytest.FixtureRequest) -> _AppResultEvent:
+    return _AppResultEvent(
+        {
+            WriterEvent.ENDPOINT_ID: "some-ep-id",
+            WriterEvent.START_INFER_TIME: "2023-09-19 14:26:06.501084",
+            WriterEvent.END_INFER_TIME: "2023-09-19 16:26:06.501084",
+            WriterEvent.APPLICATION_NAME: "dummy-app",
+            WriterEvent.RESULT_NAME: "data-drift-0",
+            WriterEvent.RESULT_KIND: 0,
+            WriterEvent.RESULT_VALUE: 0.32,
+            WriterEvent.RESULT_STATUS: request.param,
+            WriterEvent.RESULT_EXTRA_DATA: "",
+        }
+    )
```

Comment:
Nit:
You don't use parametrization, so remove this:
```suggestion
@pytest.fixture()
def event(request: pytest.FixtureRequest) -> _AppResultEvent:
    return _AppResultEvent(
        {
            WriterEvent.ENDPOINT_ID: "some-ep-id",
            WriterEvent.START_INFER_TIME: "2023-09-19 14:26:06.501084",
            WriterEvent.END_INFER_TIME: "2023-09-19 16:26:06.501084",
            WriterEvent.APPLICATION_NAME: "dummy-app",
            WriterEvent.RESULT_NAME: "data-drift-0",
            WriterEvent.RESULT_KIND: 0,
            WriterEvent.RESULT_VALUE: 0.32,
            WriterEvent.RESULT_STATUS: 0,
            WriterEvent.RESULT_EXTRA_DATA: "",
        }
    )
```